force-app/main/default/classes/SOQL.cls

```diff
@@ -1180,11 +1180,19 @@ public virtual inherited sharing class SOQL implements Queryable {
         }
 
         private Filter set(String comperator, Object value) {
+            if (value instanceof String && isDateLiteral((String) value)) {
```

Comment:
Ogólnie rozwiązanie wygląda spoko, jest krótsze niż myślałem, ale!

Widzę w tym kilka problemów jak już wspominałem:
1. Każdy filtr na orgu używa metody `set`, więc wszystkie value będą sprawdzane czy są Stringiem i czy są date literal. Dodajemy więc konkretną logikę (`if`) do generycznej metod.
2. Dodając tego ifa, który ma sens tylko dla 5% przypadków pogarszamy performance (mimo, że niewiele, ale wciąż) dla 95% pozostałych query.
3. `SOQL.Filter.with(Task.Name).equal('today')`, to uzna to jako date literal i się nie zbinduje. To jest o tyle problematyczne, że kod przestanie działać, bo String nie będzie w `' '` ani bindowany.
4. `SOQL.Filter.with(Task.Name).equal('today:bleble:costam')`. Może mam taką konwencję nazw na orgu. Uzna to jako date literal i się nie zbinduje. To jest o tyle problematyczne, że kod przestanie działać, bo String nie będzie w `' '` ani bindowany.